### PR TITLE
update with additional resource provider required for subscription microsoft.Network

### PR DESCRIPTION
**Pull request by denitiu**
Enhance 

"/area" identifiers:    documentation
"/kind" identifiers:    cleanup

Fixing the below error message, when trying to create a new cluster:
```
resource "azurerm_network_security_group" "workers" {
* creating/updating Virtual Network: (Name "shoot--mlfprod--preprod-we" / Resource Group "shoot--mlfprod--preprod-we"): network.VirtualNetworksClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="MissingSubscriptionRegistration" Message="The subscription is not registered to use namespace 'Microsoft.Network'. See https://aka.ms/rps-not-found for how to register subscriptions." Details=[{"code":"MissingSubscriptionRegistration","message":"The subscription is not registered to use namespace 'Microsoft.Network'. See https://aka.ms/rps-not-found for how to register subscriptions.","target":"Microsoft.Network"}]
  with azurerm_virtual_network.vnet,
  on main.tf line 24, in resource "azurerm_virtual_network" "vnet":
  24: resource "azurerm_virtual_network" "vnet" {
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @denitiu for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md [hunk 1]
```diff
@@ -71,6 +71,12 @@ Gardener allows you to create a Kubernetes cluster on different infrastructure p
     >After completing these steps, you should see your newly created secret in the *Infrastructure Secrets* section.
 
     <img src="images/secret-stored.png">
+    
+1. Register microsoft.Network for your subscription.
```
Comment (dkistner) [inline docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md:75]: Thanks for adding this section. I guess there will be more Azure providers needed to create a Shoot cluster e.g. `microsoft.Compute`.

Comment (kon-angelo) [reply docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md:75]: I think we can add `compute` to this guide and we can update it later with a full list.

Comment (denitiu) [reply docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md:75]: can very well be the case. Maybe I should make that section a bit more generic? 
For now, this was the only error that I received when trying to create a brand new cluster.

Comment (kon-angelo) [reply docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md:75]: I think the title can be generic like `register microsoft resource providers` and replace the last step with a list where we can add all the necessary providers.
If you didn't get another error its likely due to default settings.

Comment (denitiu) [reply docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md]: pushed a change now

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Review by dkistner — APPROVED** [review]

**Pull request merged**